.github/workflows/ci.yaml

```
@@ -79,6 +79,98 @@ jobs:
+          - "pytest_max"
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Fetch sources
+        uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: linux/${{ matrix.arch }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      # This is a workaround to prevent the cache from growing indefinitely.
+      # https://docs.docker.com/build/ci/github-actions/cache/#local-cache
+      # https://github.com/docker/build-push-action/issues/252
+      # https://github.com/moby/buildkit/issues/1896
+      - name: Cache container layers
+        uses: actions/cache@v3
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-nox-${{ matrix.arch }}-${{ matrix.os }}-${{ matrix.python }}
+
+      - name: Build image
+        uses: docker/build-push-action@v4
+        with:
+          context: .github/containers/nox-cross-arch
+          file: .github/containers/nox-cross-arch/${{ matrix.arch }}-${{ matrix.os }}-python-${{ matrix.python }}.Dockerfile
+          platforms: linux/${{ matrix.arch }}
+          tags: localhost/nox-cross-arch:latest
+          push: false
+          load: true
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache-new,mode=max
+
+      # Refer to the workaround mentioned above
+      - name: Move cache
+        run: |
+          rm -rf /tmp/.buildx-cache
+          mv /tmp/.buildx-cache-new /tmp/.buildx-cache
+
+      # Cache pip downloads
+      - name: Cache pip downloads
+        uses: actions/cache@v3
+        with:
+          path: /tmp/pip-cache
+          key: nox-${{ matrix.nox-session }}-${{ matrix.arch }}-${{ matrix.os }}-${{ matrix.python }}-${{ hashFiles('**/pyproject.toml') }}
+
+      # This ensures that the docker container has access to the pip cache.
+      # Changing the user in the docker-run step causes it to fail due to
+      # incorrect permissions. Setting the ownership of the pip cache to root
+      # before running is a workaround to this issue.
+      - name: Set pip cache owners to root for docker
+        run: if [[ -e /tmp/pip-cache ]]; then sudo chown -R root:root /tmp/pip-cache; fi
+
+      - name: Run nox
+        run: |
+          docker run \
+            --rm \
+            -v $(pwd):/${{ github.workspace }} \
+            -v /tmp/pip-cache:/root/.cache/pip \
+            -w ${{ github.workspace }} \
+            --net=host \
+            --platform linux/${{ matrix.arch }} \
+            localhost/nox-cross-arch:latest \
+            bash -c "pip install -e .[dev-noxfile]; nox --install-only -e ${{ matrix.nox-session }}; pip freeze; nox -e ${{ matrix.nox-session }}"
+        timeout-minutes: 30
+
+      # This ensures that the runner has access to the pip cache.
+      - name: Reset pip cache ownership
+        run: sudo chown -R $USER:$USER /tmp/pip-cache
```

If the previous step fails, then the ownership reset won't be done, right? Shouldn't this use [`if: always()`](https://docs.github.com/en/actions/learn-github-actions/expressions#always)?
```suggestion
      # This ensures that the runner has access to the pip cache.
      - name: Reset pip cache ownership
        if: always()
        run: sudo chown -R $USER:$USER /tmp/pip-cache
```

---

True, makes sense. Done